crates/rethnet_evm/src/state/fork.rs

```diff
@@ -176,7 +196,16 @@ impl StateDebug for ForkState {
     }
 
     fn remove_account(&mut self, address: Address) -> Result<Option<AccountInfo>, Self::Error> {
-        self.local_state.remove_account(address)
+        if let Some(account_info) = self.local_state.remove_account(address)? {
```

Comment:
It's slightly more intricate, I think. I expect the following test to still fail with this code:

0. remote account `A` exists
1. modify account `A`, which creates a local copy
2. remove account `A`
3. calling `basic()` will now return the remote account `A`

To solve this, follow a similar pattern to `set_account_storage_slot`:
1. `self.removed_remote_accounts.insert(address);`
2. call and return `self.local_state_remove_account(address)`

There is one more intricacy: dealing with `set_block_context`. To handle those, you'll need to add a function similar to `update_removed_storage_slots()` for updating removed accounts and call that at the end of `set_block_context`